fix(app): show correct git status for newly created/modified files

lmiller1990 opened the pull request:
<!-- Thanks for contributing! PLEASE...
- Read our contributing guidelines: https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md 
- Read our Code Review Checklist on coding standards and what needs to be done before a PR can be merged: https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#Code-Review-Checklist
- Mark this PR as "Draft" if it is not ready for review.
- Make sure you set the correct base branch based on what packages you're changing: https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#branches
-->

- Closes https://cypress-io.atlassian.net/browse/UNIFY-863

### User facing changelog
<!-- 
Explain the change(s) for every user to read in our changelog. Examples: https://on.cypress.io/changelog
If the change is not user-facing, write "n/a".
-->

- Show correct git status for untracked, staged and modified files
- Align UI with Figma mocks (Should not show git status on folders, just files)
- Fix bug where git status is not appearing on linux (git command not working in `/bin/sh`; default to `$SHELL` or `/bin/bash` for unix-like systems).
- Add proper tests (create an actual git repo; not mocking the response)

![image](https://user-images.githubusercontent.com/19196536/159637308-a7778ad8-68b1-4eb4-b634-943ee5b9603e.png)


### Additional details
<!-- Examples:
- Why was this change necessary?
- What is affected by this change?
- Any implementation details to explain?
-->

N/A

### How has the user experience changed?
<!-- Provide before and after examples of the change.
Screenshots or GIFs are preferred. -->

### PR Tasks
<!-- 
These tasks must be completed before a PR is merged.
If a task does not apply, write [na] instead of checking the box.
DO NOT DELETE the PR checklist.
-->

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? <!-- Link to PR here -->
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

lmiller1990 commented on packages/data-context/test/unit/sources/GitDataSource.spec.ts:
```diff
@@ -0,0 +1,59 @@
+import { createTestDataContext, scaffoldMigrationProject } from '../helper'
+
+describe('GitDataSource', () => {
+  it(`gets correct status for files on ${os.platform()}`, async () => {
```
Making sure it runs on windows/linux/macos.

marktnoonan reviewed (changes requested):
I've done a bit of testing and seems to work great overall. There are two things to look at. First, when creating a spec from Cypress, it doesn't show the git status in the table, see `filename.cy.ts` here:

![Screen Shot 2022-03-24 at 1 44 27 PM](https://user-images.githubusercontent.com/8340719/159979195-8cf72c9d-a05d-4c75-a483-cdafcb6badd5.png)

Second, in the mocks, before files are committed, the designs show "created" and "modified" without the name attached. Should we match that here?

Love the new tests!

elevatebart commented on packages/types/src/git.ts:
```diff
@@ -0,0 +1 @@
+export const gitStatusType = ['modified', 'created', 'unmodified'] as const
```
It's almost like we should have this model of files for everything in the `constants.ts` file.

I like this though.

lmiller1990 commented:
@marktnoonan I have

- changed UI not to show username for created/modified
- fixed bugs where UI is not updating when you create a new spec from the UI
- other small bugs I found

I put a video demo here to show the new UI and it working as expected: https://cypress-io.atlassian.net/browse/UNIFY-863?focusedCommentId=17183

This was a surprisingly complex ticket. Our virtualized spec list is a bit weird (well, the API with `useSpecList`). We should revisit this - it was the cause of the UI not updating correctly, it was doing some caching, preventing the UI from updating.

marktnoonan commented:
@lmiller1990 I'm confused by this Percy [snapshot](https://percy.io/cypress-io/cypress/builds/16817567/changed/945431778?browser=chrome&browser_ids=20&subcategories=unreviewed%2Cchanges_requested&viewLayout=overlay&viewMode=new&width=800&widths=400%2C800%2C1000%2C1024), and [this one](https://percy.io/cypress-io/cypress/builds/16818194/changed/945471300?browser=chrome&browser_ids=20&subcategories=unreviewed%2Cchanges_requested&viewLayout=overlay&viewMode=new&width=800&widths=400%2C800%2C1000%2C1024).

The first seems to show results where we should be on the empty state, and both show `file.git.null` in the UI. I am OOO the rest of the day and haven't looked any deeper into this. I've dismissed my change request since I won't be able to re-review until Monday.

I'd agree about revisiting the specs list, @ZachJW34 and I have talked about this as well. There are a couple of related problems it would be good to talk about together.

ryanthemanuel commented:
Should foo.cy.js and blah.cy.js have been checked in?  It looks like they're generated as part of a test

lmiller1990 commented:
@marktnoonan thanks for catching this. I'd forgotten to update the stubs, in [this file](https://github.com/cypress-io/cypress/blob/10.0-release/packages/frontend-shared/cypress/support/mock-graphql/testStubSpecs.ts).

It's fixed now. I hard coded "modified 1 minute ago" since I think it makes it easier to search for. Now Percy should no longer show `git.info.null`. We will need to re-approve the status. I think we should move away from the highly dynamic fixtures we have, it's quite hard to debug and write assertions against, since you need to run the test to see what Faker will generate.

Happy to wait for your re-review if you are out, or get someone else to :eyes:.

@ryanthemanuel thanks, I removed those two junk testing files.

@elevatebart Can you please re-review?
